### PR TITLE
Cortical borers can not invade your head if your head is fully bio protected

### PR DESCRIPTION
**Pull request by Paxilmaniac**
## About The Pull Request

As the title might imply, if you have max bio armor on your head, be it a helmet, mask, whatever, then a borer cannot enter you.

## How This Contributes To The Skyrat Roleplay Experience

Borers are currently basically a "lol get fucked" if one decides it wants to enter you. This is because there are no means of protecting yourself from a borer. With how stupidly dangerous cortical borers can be, I don't think we should leave them without a sensical counter to them entering your head.

## Proof of Testing

Not yet

## Changelog

:cl:
balance: Any helmet or mask that has 100 bio armor (Biological armor level X for the tag fans) will protect you against cortical borers entering your head.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 1]
```diff
@@ -548,6 +548,9 @@
 
 /datum/action/cooldown/borer/choosing_host/proc/enter_host(mob/living/carbon/human/singular_host)
 	var/mob/living/basic/cortical_borer/cortical_owner = owner
+	if(check_for_bio_protection(singular_host))
+		owner.balloon_alert(owner, "target head too protected")
```
Comment (tf-4) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]:
```suggestion
		owner.balloon_alert(owner, "target head too protected!")
```
Balloon alerts for fail states use `!`

**Review by tf-4 — CHANGES_REQUESTED** [review]

**Review by tf-4 — APPROVED** [review]

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 2]
```diff
@@ -571,6 +574,19 @@
 	cortical_owner.human_host.log_message(logging_text, LOG_GAME)
 	StartCooldown()
 
+/// Checks if the target's head is bio protected, returns true if this is the case
+/datum/action/cooldown/borer/choosing_host/proc/check_for_bio_protection(mob/living/carbon/human/target)
+	if(isobj(target.head))
+		if(target.head.get_armor_rating(BIO) == 100)
```
Comment (Zonespace27) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]:
```suggestion
		if(target.head.get_armor_rating(BIO) >= 100)
```

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 3]
```diff
@@ -571,6 +574,19 @@
...
+			return TRUE
+	if(isobj(target.wear_mask))
+		if(target.wear_mask.get_armor_rating(BIO) == 100)
```
Comment (Zonespace27) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]:
```suggestion
		if(target.wear_mask.get_armor_rating(BIO) >= 100)
```

File: modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm [hunk 4]
```diff
@@ -571,6 +574,19 @@
...
+			return TRUE
+	if(isobj(target.wear_neck))
+		if(target.wear_neck.get_armor_rating(BIO) == 100)
```
Comment (Zonespace27) [inline modular_skyrat/modules/cortical_borer/code/cortical_borer_abilities.dm]:
```suggestion
		if(target.wear_neck.get_armor_rating(BIO) >= 100)
```

**Review by Zonespace27 — APPROVED** [review]
Lgtm

**Pull request merged**